yarn.lock

```diff
@@ -5118,14 +4960,8 @@ ansi-align@^2.0.0:
   dependencies:
     string-width "^2.0.0"
 
-ansi-align@^3.0.0:
-  version "3.0.0"
-  resolved "https://registry.yarnpkg.com/ansi-align/-/ansi-align-3.0.0.tgz#b536b371cf687caaef236c18d3e21fe3797467cb"
-  integrity sha512-ZpClVKqXN3RGBmKibdfWzqCY4lnjEuoNzU5T0oEFpfd/z5qJHVarukridD4juLO2FXMiwUQxr9WqQtaYa8XRYw==
-  dependencies:
-    string-width "^3.0.0"
-
-ansi-colors@3.2.3:
+ansi-align@^3.0.0, ansi-colors@3.2.3:
+  name ansi-align
```

Comment:
@ecraig12345 If this was a manual change I believe it's broken the repo from being able to be built. I'll need to try and revert (at least) this change. Were there any other manual changes to the lockfile being made as part of this? Is manually modifying the yarn.lock file a common practice? My concern would be the introduction of breaks fairly easily be messing with how yarn resolves and manages dependencies.

Reply:
Yeah...that part was a mistake, sorry. Yarn probably warned about it but it got lost in the sea of [spurious missing peer dependency warnings](https://github.com/yarnpkg/yarn/issues/5810). I just pushed a fix. (The fix process is to delete the invalid entry and run `yarn` to let it be regenerated.)

To the other part of your question, there were quite a few manual changes because yarn tends to resolve new dependencies in a way that results in often-unnecessary duplicates (like we saw with babel and storybook), and I'm not sure how to completely work around it without manual changes.

For example, say you have some existing (probably indirect) dependency on `foo@^1.1.0` which yarn resolved to `foo@1.1.0` and put in the lockfile. Then something else adds a dependency (again, probably indirect) on `foo@^1.2.0` . Instead of changing both to resolve to `foo@1.2.0`, which satisfies both semver specs, it adds a *separate* entry/resolution just for the new spec. (IIRC this can even happen with *older* semver specs: a new indirect dep on `foo@^1.0.0` will also result in a separate resolution being added to the *newer* `foo@1.2.0`.)

I'm pretty sure this is an intentional behavior done for safety purposes (at least avoiding the implicit upgrade of the original `foo@^1.1.0` to `foo@1.2.0`...adding a separate resolution to a newer version for `foo@^1.0.0` just seems weird) but is less than ideal from an install time/size perspective--especially in cases like babel or storybook stuff that has a bunch of dependencies and you get a cascading effect.

Regarding how to validate manual changes, yarn does warn on blatantly invalid things like this, so I should have looked more closely through the sea of peer dep errors to check for any real issues. Beyond that, generally as long as the repo builds and tests pass afterwards, it's probably fine. Also I try and avoid messing with direct dependencies where version upgrades may be particularly impactful, but we usually try and lock those down to a pretty specific version (`~` or exact) to begin with.

Reply:
It's probably possible to resolve some of this without manual changes using [`yarn upgrade`](https://classic.yarnpkg.com/en/docs/cli/upgrade) or [`yarn upgrade-interactive`](https://classic.yarnpkg.com/en/docs/cli/upgrade-interactive), and maybe I should try that first in the future instead. Both take an optional package name or scope, so it may be possible to do something similar to what I was doing manually with `yarn upgrade-interactive`.